tests/network/primary_pod_network.go

```diff
@@ -178,15 +217,34 @@ func vmiWithMasqueradeBinding() *v1.VirtualMachineInstance {
 	return vmi
 }
 
-func newFedoraWithGuestAgent() (*v1.VirtualMachineInstance, error) {
+func newFedoraWithGuestAgentAndDefaultInterface(iface v1.Interface) (*v1.VirtualMachineInstance, error) {
 	networkData, err := libnet.CreateDefaultCloudInitNetworkData()
 	if err != nil {
 		return nil, err
 	}
 
-	tmpVmi := libvmi.NewFedora(
+	vmi := libvmi.NewFedora(
+		libvmi.WithInterface(iface),
+		libvmi.WithNetwork(v1.DefaultPodNetwork()),
 		libvmi.WithCloudInitNoCloudUserData(tests.GetGuestAgentUserData(), false),
 		libvmi.WithCloudInitNoCloudNetworkData(networkData, false),
 	)
-	return tmpVmi, nil
+	return vmi, nil
+}
+
+func removeCIDR(ip string) string {
+	ipAddr, _, err := net.ParseCIDR(ip)
+	if err != nil {
```

Comment:
aint it too broad ?
I understand the comparison in the Eventually will catch errors, but still

Reply:
It looks fine to me. If the IP cannot be parsed as a CIDR, for whatever reason, it is returned as is and at a different place it may explode (e.g. due to bad IP format).